ledger/sandbox/src/main/scala/com/digitalasset/platform/store/CompletionFromTransaction.scala

```diff
@@ -3,29 +3,68 @@
 
 package com.digitalasset.platform.store
 
+import java.time.Instant
+
 import com.daml.ledger.participant.state.v1.TransactionId
+import com.digitalasset.api.util.TimestampConversion.fromInstant
 import com.digitalasset.daml.lf.data.Ref
 import com.digitalasset.ledger.{ApplicationId, CommandId}
 import com.digitalasset.ledger.api.domain
-import com.digitalasset.ledger.api.domain.CompletionEvent
-import com.digitalasset.ledger.api.domain.CompletionEvent.{CommandAccepted, CommandRejected}
+import com.digitalasset.ledger.api.domain.RejectionReason
+import com.digitalasset.ledger.api.domain.RejectionReason.{
+  Disputed,
+  Inconsistent,
+  OutOfQuota,
+  PartyNotKnownOnLedger,
+  SubmitterCannotActViaParticipant,
+  TimedOut
+}
+import com.digitalasset.ledger.api.v1.command_completion_service.{
+  Checkpoint,
+  CompletionStreamResponse
+}
+import com.digitalasset.ledger.api.v1.completion.Completion
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
 import com.digitalasset.platform.store.dao.LedgerDao
 import com.digitalasset.platform.store.entries.LedgerEntry
+import com.google.rpc.status.Status
+import io.grpc.Status.Code
 
 // Turn a stream of transactions into a stream of completions for a given application and set of parties
 // TODO Remove this when:
 // TODO - the participant can read completions off the index directly AND
 // TODO - the in-memory sandbox is gone
 private[platform] object CompletionFromTransaction {
 
-  private def toApiOffset(offset: LedgerDao#LedgerOffset): domain.LedgerOffset.Absolute =
-    domain.LedgerOffset.Absolute(Ref.LedgerString.assertFromString(offset.toString))
+  private def toApiCheckpoint(
+      recordedAt: Instant,
+      offset: LedgerDao#LedgerOffset): Option[Checkpoint] =
+    Option(
```

Comment:
I would use `Some` here and lower down when it's clear it's a valid object, but not going to complain too much. I'd like to see consistency.

Reply:
Makes sense, thanks.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4675/commits/69d2c05532d6197301717fba2a458b909a855af8